### PR TITLE
feat: complete profile fifth form FE

### PR DESCRIPTION
**Pull request by nadiia-tsytsylina**
## Describe your changes

create Complete profile fifth component

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/AC-73l)

Снимок экрана 2023-11-23 134328
Снимок экрана 2023-11-23 134337
Снимок экрана 2023-11-23 141429
Снимок экрана 2023-11-23 141437

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/components/complete-profile-fourth/CompleteProfileFifth.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    const marks = [
+      {
+        value: 0,
+        label: '0',
+      },
+      {
+        value: 200,
+        label: '200',
+      },
+    ];
```
Comment (lyudad) [inline src/components/complete-profile-fourth/CompleteProfileFifth.tsx:35]: it will reinialise with each rerender

Comment (nadiia-tsytsylina) [reply src/components/complete-profile-fourth/CompleteProfileFifth.tsx]: move it to constants

File: src/components/complete-profile-fifth/CompleteProfileFifth.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          name="rate"
+          control={control}
+          render={({ field }): JSX.Element => (
+            <Slider
+              {...field}
+              size="medium"
+              onChange={(value): void => field.onChange(value)}
+              aria-labelledby="input-slider"
+              min={0}
+              max={200}
```
Comment (Lilly-nn) [inline src/components/complete-profile-fifth/CompleteProfileFifth.tsx:56]: Would be nice if max value was took from constants

File: src/components/complete-profile-fifth/CompleteProfileFifth.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                </InputLabel>
+
+                <FilledInput
+                  {...field}
+                  error={!!errors.rate}
+                  onChange={(value): void => field.onChange(value)}
+                  inputProps={{
+                    step: 1,
+                    min: 0,
+                    max: 200,
```
Comment (Lilly-nn) [inline src/components/complete-profile-fifth/CompleteProfileFifth.tsx:79]: same here

File: src/components/complete-profile-fifth/CompleteProfileFifth.tsx (hunk 2 above)
Comment (nadiia-tsytsylina) [reply src/components/complete-profile-fifth/CompleteProfileFifth.tsx]: thanks, fixed

File: src/components/complete-profile-fifth/CompleteProfileFifth.tsx (hunk 3 above)
Comment (nadiia-tsytsylina) [reply src/components/complete-profile-fifth/CompleteProfileFifth.tsx]: fixed

File: src/components/complete-profile-fifth/validation.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    rate: undefined;
+  },
+  ''
+> => {
+  const { translate } = useLocales();
+
+  return object({
+    rate: number()
+      .min(1, translate('completeProfileFifth.minRateError'))
+      .max(200, translate('completeProfileFifth.maxRateError'))
```
Comment (Lilly-nn) [inline src/components/complete-profile-fifth/validation.ts:21]: and here too

Comment (nadiia-tsytsylina) [reply src/components/complete-profile-fifth/validation.ts]: fixed

**Review by ProggerOleg — APPROVED** [review]

File: src/locales/langs/en.ts [hunk 5]
```diff
@@ -165,6 +165,12 @@ const en = {
     housekeeping: 'Housekeeping and Laundry',
     socialActivities: 'Social and Recreational Activities',
   },
+  completeProfileFifth: {
+    placeholderRate: 'Rate ($/h)',
+    minRateError: 'minimal value is 1 $/h',
+    maxRateError: 'maximal value is 200 $/h',
```
Comment (fedasevich) [inline src/locales/langs/en.ts:171]: Maybe lets make it interpolated value

**Pull request merged**